website/content/docs/custom-widgets.md

```diff
@@ -198,3 +195,276 @@ You can also return a promise from `isValid`. While the promise is pending, the
+       rules: [
+         {
+           test: /\.js$/,
+           loader: 'source-map-loader',
+           enforce: 'pre',
+         },
+         {
+           test: /\.jsx?$/,
+           exclude: /node_modules/,
+           loader: 'babel-loader',
+         },
+         {
+           test: /\.css$/,
+           loader: ['style-loader', 'css-loader'],
+         },
+       ],
+     },
+     plugins: [
+       new HtmlWebpackPlugin(),
+     ],
+     devtool: 'eval-source-map',
+   }
+
+   const productionConfig = {
+     mode: 'production',
+     module: {
+       rules: [
+         {
+           test: /\.jsx?$/,
+           loader: 'babel-loader',
+         },
+       ],
+     },
+     devtool: 'source-map',
+   }
+
+   module.exports = process.env.NODE_ENV === 'production' ? productionConfig : developmentConfig
+   ```
+8. For installing the dependencies run this command:
+
+   ```javascript
+   npm install
+   ```
+
+### [](https://github.com/netlify/netlify-cms-widget-starter#development)Development
+
+To run a copy of Netlify CMS with your widget for development, use the start script:
+
+```javascript
+npm start
+```
+
+Your widget source is in the `src` directory, where there are separate files for the `Control` and `Preview` components.
+
+### [](https://github.com/netlify/netlify-cms-widget-starter#production--publishing)Production & Publishing
+
+You'll want to take a few steps before publishing a production built package to npm:
+
+1. Customize `package.json` with details for your specific widget, e.g. name, description, author, version, etc.
+
+   ```json
+   {
+     "name": "netlify-cms-widget-starter",
+     "description": "A boilerplate for creating Netlify CMS widgets.",
+     "author": "name of developer",
+     "keywords": [
+       "netlify",
+       "netlify-cms",
+       "cms",
+       "widget",
+       "starter",
+       "boilerplate"
+     ],
+     "version": "0.0.1",
+     // ... rest
+   }
+   ```
+2. For discoverability, ensure that your package name follows the pattern `netlify-cms-widget-<name>`.
+3. Delete this `README.md`, rename `README_TEMPLATE.md` to `README.md`, and update the new file for your specific widget.
+4. Rename the exports in `src/index.js`. For example, if your widget is `netlify-cms-widget-awesome`, you would do:
+
+```javascript
+if (typeof window !== 'undefined') {
+  window.AwesomeControl = Control
+  window.AwesomePreview = Preview
+}
+
+export { Control as AwesomeControl, Preview as AwesomePreview }
+```
+
+5. Optional: customize the component and file names in `src`.
+6. If you haven't already, push your repo to your GitHub account so the source available to other developers.
+7. Create a production build, which will be output to `dist`:
+
+```javascript
+npm run build
+```
+
+8. Finally, if you're sure things are tested and working, publish!
+
+```javascript
+npm publish
+```
+
+### [](https://github.com/netlify/netlify-cms-widget-starter#deploying-a-live-demo)Deploying a live demo
```

Comment:
We can drop this part as we're no longer in the context of a git repo.

Reply:
I have dropped it, fyi